### ISSUE TITLE
[HOLD for payment 2022-12-20]  Android/IOS- The flow to the transaction amount fields does not work in apps

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#11068]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Issue found when executing PR** https://github.com/Expensify/App/pull/10838

## Action Performed:
1. Open the App and login with any account
2. Go to Settings->Workspace->Connect bank account->Connect manually
3. Enter the details to add an account

## Expected Result:
The flow to the transaction amount fields tab should occur

## Actual Result:
The flow to the transaction amount fields does not work in apps

## Workaround:
Unknown

## Platform:

Where is this issue occurring?

- iOS
- Android

**Version Number:** 1.2.1.0

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**Email or phone of affected tester (no customers):** any

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/190690149-4f0e2f0c-55cd-4c7b-bab2-70a93fe2ea44.mp4

https://user-images.githubusercontent.com/93399543/190690207-59220673-b125-4a56-abb0-edd15f89b726.mp4

**Expensify/Expensify Issue URL:** 

**Issue reported by:** Applause - Internal Team

**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by sketchydroide** [comment]
I'm not sure this is a bug or a feature, that looks like a security feature, not sure if it should be active for a test account, but it seems like the kind of thing that sould sometimes happen.

@marcaaron @MariaHCD maybe one of you know more about this, or can direct to someone that does?

**Comment by MariaHCD** [comment]
As I understand it, on production, Onfido should be the next step after the personal information form (source: [SO](https://stackoverflow.com/c/expensify/questions/342))

However, when adding the test bank account on staging, we skip Onfido (source code [here](https://github.com/Expensify/Web-Secure/blob/7bf5057c192dcff46fd88db1f0f99ba0a5143835/lib/BankAccount/VerificationAPI.php#L560-L572))

So:
- On production, we won't skip the Onfido check
- On staging, we'll skip the Onfido check if the test bank account numbers are used

@sketchydroide I think you'll just need to confirm why Onfido wasn't skipped in the first recording on the App because it looks like the staging server option was enabled.

**Comment by sketchydroide** [comment]
thanks, let me check that

**Comment by sketchydroide** [comment]
It seems that the switch only sets the URL to stagin or not, there is no other logic on the app itself, going to check Web now

**Comment by sketchydroide** [comment]
I still have not been able to get to the bottom of it.

**Comment by sketchydroide** [comment]
this is not particularly urgent setting it as weekly, and will continue to look into it

**Comment by sketchydroide** [comment]
> On staging, we'll skip the Onfido check if the test bank account numbers are used

I found on Web that 'isPlaidTestRequestor' is used with the card number the QA used.
But this is on Web, I have no idea why this would not work on the app

**Comment by sketchydroide** [comment]
I have found no evidence of us making this test on the App

**Comment by sketchydroide** [comment]
@kbecciv is this still an issue?

**Comment by kbecciv** [comment]
Yes, we still blocked for testing in apps

**Comment by sketchydroide** [comment]
looking again this week

**Comment by sketchydroide** [comment]
I'm removing myself, I have not been able to look at this, and I'm not sure I will, I'm leaving this so that someone can pick it up.

**Comment by MariaHCD** [comment]
I think this issue might possibly be a bit stale, we've made improvements/refactors to the VBBA flow in the past month so it's worth retesting this on staging to see if it's still occurring. cc: @kbecciv

**Comment by kbecciv** [comment]
The flow has different behavior now, 

https://user-images.githubusercontent.com/93399543/200032273-0f834279-de1e-46a4-bf5f-a253fca56b93.mp4

**Comment by MariaHCD** [comment]
Thanks, @kbecciv. It looks like trying to set up a PENDING bank account created a VERIFYING bank account instead (cc: @nkuoch)

@kbecciv can you confirm if this is happening on web and desktop or if this is only happening on Android and iOS.

**Comment by muttmuure** [comment]
I'm going to hold for @kbecciv - I'm still not sure what the bug is here. When you've confirmed please can you update the OP and reassign me

**Comment by kbecciv** [comment]
@muttmuure Checking in Web and Desktop, will update you shortly

**Comment by kbecciv** [comment]
Issue is not reproduced on Web nor Desktop app, build 1.2.26.0

https://user-images.githubusercontent.com/93399543/201205045-48c7d8a7-1bfa-4c0f-8fa5-28b41c63c1dc.mp4


https://user-images.githubusercontent.com/93399543/201205102-40af7462-cd48-4b92-bab9-5718827401a0.mp4

**Comment by MariaHCD** [comment]
Thanks for confirming, @kbecciv. Looks like the issue is happening on mobile but not on web or desktop. Not sure what could be causing the discrepancy between mobile and web platforms since the information entered on mobile matches the data in the [SO](https://stackoverflow.com/c/expensify/questions/342) to create a verifying bank account. I think we're good to apply the engineering label again here, @muttmuure.

**Comment by muttmuure** [comment]
Didn't realise this was still open. Going to retest this

**Comment by muttmuure** [comment]
Can't get the manual flow https://expensify.slack.com/archives/C01GTK53T8Q/p1669637620523379

**Comment by muttmuure** [comment]
OK this is still happening for apps.

TLDR; We need to add to add 'isPlaidTestRequestor'  to iOS and Android to skip Onfido so Applause can test adding a VBA in apps.

**Comment by marcochavezf** [comment]
I chatted with @joelbettner that I'm going to grab this one. Working on it today

**Comment by marcochavezf** [comment]
After investigating a bit further, I noticed in the logs that [isPlaidTestRequestor](https://github.com/Expensify/Web-Expensify/blob/68f38648d73d26c14d6cb264fa3fb3502b450301/lib/BankAccount/VerificationAPI.php#L195) returns false (and thus `markStepsAsVerified` is not called). This happens because the Staging app is not pointing to the Staging API, and seems the reason is that `CONFIG.IS_IN_STAGING` is false [here](https://github.com/Expensify/App/blob/8e3cac5483e2afd2a06491ab6a5acaed344a833d/src/libs/HttpUtils.js#L101), but it would be fixed in this PR https://github.com/Expensify/App/pull/13029

**Comment by marcochavezf** [comment]
The [PR](https://github.com/Expensify/App/pull/13029) to fix this issue is in review.

**Comment by marcochavezf** [comment]
Waiting for https://github.com/Expensify/App/pull/13029 to be deployed to staging

**Comment by muttmuure** [comment]
@marcochavezf are we still waiting on this one to be merged? It doesn't sound like there's a bug?

**Comment by marcochavezf** [comment]
HI @muttmuure, the fix was already deployed to production, I tested it with the staging app (Android), and looks good. 

Test steps:

1. Verify that in the Preferences menu, the option "User Staging Server" is on (otherwise the Onfido step won't be skipped). 
2. Follow the instructions to add a PENDING account.
3. Verify the Onfido step is skipped after step 3 and that the page "Test transactions" is shown (page to validate the three amounts).

**Comment by muttmuure** [comment]
I'm playing catchup due to being out for the holiday and moving house. I'll get this done as I get through my K2

**Comment by muttmuure** [comment]
I'm adding to an existing step that Applause are completing anywhere. Detailed here: https://github.com/Expensify/Expensify/issues/255238

**Comment by muttmuure** [comment]
Thanks @marcochavezf!
